PubHub/PubHub.API/Controllers/UserController.cs

```diff
@@ -142,23 +143,91 @@ public async Task<IActionResult> DeleteUserAsync(int id)
                 .SingleOrDefaultAsync();
             if (accountId == default)
             {
-                return Problem(
+                return Results.Problem(
                     statusCode: StatusCodes.Status404NotFound,
                     detail: $"No user with ID: {id}");
             }
 
             // Delete account.
-            int updatedRows = await _context.Users
+            int updatedRows = await _context.Set<Account>()
                 .Where(u => u.Id == accountId)
                 .ExecuteUpdateAsync(u => u.SetProperty(u => u.IsDeleted, true));
             if (updatedRows > 0)
             {
-                return Ok();
+                return Results.Ok();
             }
 
-            return Problem(
+            // Unable to delete; report back.
+            Dictionary<string, object?> extensions = new()
+            {
+                ["UserId"] = id,
+                ["AccountId"] = accountId
+            };
+
+            return Results.Problem(
+                statusCode: StatusCodes.Status500InternalServerError,
+                detail: $"Unable to delete account (ID: {accountId}) of user (ID: {id}).",
+                extensions: extensions);
```

Comment:
Is extensions the same as instance, or is my understanding flawed here?

Reply:
I think they are different. Both `instance` and `extensions` can be set at the same time and `instance` is of type `string?`, whereas `extensions` is a dictionary. 🤔

Reply:
Interesting. Might need to read up on th especification 👌